packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -377,17 +375,17 @@ class _SnackBarState extends State<SnackBar> {
         break;
       case AnimationStatus.completed:
         if (widget.onVisible != null && !_wasVisible) {
-          widget.onVisible();
+          widget.onVisible!();
         }
         _wasVisible = true;
     }
   }
 
   @override
   Widget build(BuildContext context) {
-    final MediaQueryData mediaQueryData = MediaQuery.of(context);
+    final MediaQueryData mediaQueryData = MediaQuery.of(context)!;
```

Comment:
Should have a `debugCheckHasMediaQuery` call.

Reply:
added